.github/workflows/esbuild-publish.yml

```diff
@@ -0,0 +1,48 @@
+name: esbuild Compilation & npm Publish Workflow
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Set the branch to use for automation tests'
+        type: string
+        required: false
+        default: 'main'
+      nodeVersion:
+        description: version of node to use.  It's better to specify latest, lts/* or lts/-1 than to hardode numbers
+        type: string
+        default: lts/*
+        required: false
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch }}
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ inputs.nodeVersion }}
+          registry-url: 'https://registry.npmjs.org'
+      - name: Install esbuild Dependencies
+        run: |
+          yarn add esbuild@^0.19.5 -D
+          yarn add esbuild-plugin-pino@^2.1.0 -D
+          yarn add npm-dts@^1.3.12 -D
+          yarn add esbuild-plugin-tsc@^0.4.0 -D
+      - name: Install All Dependencies
+        run: |
+          yarn install --network-timeout 600000
+      - name: Update for Bundling
+        run: |
+          node scripts/updateForBundling.js
+      - name: Generate Bundle
+        run: |
+          node scripts/build.js
+      - name: Publish a Package
+        run: |
+          npm config set //registry.npmjs.org/:_authToken=$NODE_AUTH_TOKEN
```

Comment:
I don't think this will work, but if it does make sure the token isn't included in the logs. 

IIRC, we use `actions/setup-node` which writes `NODE_AUTH_TOKEN` to the npmrc file.

Reply:
I tried the .npmrc method first but it did not work properly. I am sure it won't display in the logs. Here is an example in my forked repo. https://github.com/mingxuanzhangsfdx/sfdx-core/actions/runs/7955097070/job/21713519127#step:10:1